Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchyBasedLinkResolver.swift

```diff
@@ -58,6 +57,42 @@ final class PathHierarchyBasedLinkResolver {
         }
     }
     
+    /// Returns the direct descendants of the given page that match the given source language filter.
+    ///
+    /// A descendant is included if it has a language representation in at least one of the languages in the given language filter or if the language filter is empty.
+    ///
+    /// - Parameters:
+    ///   - reference: The identifier of the page whose descendants to return.
+    ///   - languagesFilter: A set of source languages to filter descendants against.
+    /// - Returns: The references of each direct descendant that has a language representation in at least one of the given languages.
+    func directDescendants(of reference: ResolvedTopicReference, languagesFilter: Set<SourceLanguage>) -> Set<ResolvedTopicReference> {
+        guard let id = resolvedReferenceMap[reference] else { return [] }
+        let node = pathHierarchy.lookup[id]!
+        
+        func directDescendants(of node: PathHierarchy.Node) -> [ResolvedTopicReference] {
+            return node.children.flatMap { _, container in
+                container.storage.compactMap { element in
+                    guard let childID = element.node.identifier, // Don't include sparse nodes
+                          !element.node.specialBehaviors.contains(.excludeFromAutomaticCuration),
+                          languagesFilter.isEmpty || !element.node.languages.isDisjoint(with: languagesFilter)
```

Comment:
Why not use `isSuperset(of:)` here also? Like I said in my other comment, maybe we don't need to check for isEmpty here explicitly like this?

In other words, is `isSuperset(of: other)` equivalent to `other.isEmpty || !isDisjoint(with: other)` ?

Reply:
It's to support passing multiple languages as a filter and including nodes that match one of those languages. 

In this case 
 `isSuperset(of: other)` is false
 `other.isEmpty || !isDisjoint(with: other)` is true.